cmd/vault-staging-k8s-secret/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM golang:1.17.3-stretch
+
+ENV \
+ GOARCH="amd64" \
+ PATH="$PATH:/go/bin"
+
+RUN apt-get update \
+ && apt-get install -y \
+         apt-transport-https \
+         ca-certificates \
+ && echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list \
+ && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add - \
+ && apt-get update \
+ && apt-get install -y google-cloud-sdk
+
+WORKDIR /go/src/github.com/teamsnap/vault-staging-k8s-secret
+
+COPY go.mod go.mod
+
+RUN go version \
+ && go mod download \
+ && go mod verify
+
+COPY . .
+
+RUN env GOOS=linux GOARCH=amd64 go build -o vault-staging-k8s-secret
+
+CMD ./entrypoint.sh
```

Comment:
This doesn't block this PR, but do we want to add a secondary build step that is a slim OS without code and dependencies (that we copy the binary over to)?

Reply:
Took a few iterations but I think I have a decent looking dockerfile now.  

before / after 
```sh
01:58:20 PM ❯ docker image ls
REPOSITORY                                             TAG                 IMAGE ID       CREATED          SIZE
us.gcr.io/staging-205121/vault-key/staging-k8-secret   0.0.1-fat           8548c7cac49d   13 minutes ago   4.14GB
us.gcr.io/staging-205121/vault-key/staging-k8-secret   0.0.1               9762d4b4f75a   35 minutes ago   769MB

Reply:
after pushing to gcr it becomes a much slimmer file ~197 MB